core/src/main/scala/cats/instances/sortedMap.scala

```diff
@@ -0,0 +1,173 @@
+import scala.collection.immutable.SortedMap
+import scala.collection.mutable
+
+trait SortedMapInstances extends SortedMapInstances1 {
+
+  implicit def catsKernelStdHashForSortedMap[K: Hash: Order, V: Hash]: Hash[SortedMap[K, V]] =
+    new SortedMapHash[K, V]
+
+  implicit def catsKernelStdMonoidForSortedMap[K: Order, V: Semigroup]: Monoid[SortedMap[K, V]] =
+    new SortedMapMonoid[K, V]
+
+  implicit def catsStdShowForSortedMap[A: Order, B](implicit showA: Show[A], showB: Show[B]): Show[SortedMap[A, B]] =
+    new Show[SortedMap[A, B]] {
+      def show(m: SortedMap[A, B]): String =
+        m.iterator
+          .map { case (a, b) => showA.show(a) + " -> " + showB.show(b) }
+          .mkString("SortedMap(", ", ", ")")
+    }
+
+
+  // scalastyle:off method.length
+  implicit def catsStdInstancesForSortedMap[K: Order]: Traverse[SortedMap[K, ?]] with FlatMap[SortedMap[K, ?]] =
+    new Traverse[SortedMap[K, ?]] with FlatMap[SortedMap[K, ?]] {
+
+      implicit val orderingK: Ordering[K] = Order[K].toOrdering
+
+      def traverse[G[_], A, B](fa: SortedMap[K, A])(f: A => G[B])(implicit G: Applicative[G]): G[SortedMap[K, B]] = {
+        val gba: Eval[G[SortedMap[K, B]]] = Always(G.pure(SortedMap.empty(Order[K].toOrdering)))
+        Foldable.iterateRight(fa.iterator, gba){ (kv, lbuf) =>
+          G.map2Eval(f(kv._2), lbuf)({ (b, buf) => buf + (kv._1 -> b)})
+        }.value
+      }
+
+      def flatMap[A, B](fa: SortedMap[K,A])(f: A => SortedMap[K,B]): SortedMap[K,B] =
+        fa.flatMap { case (k, a) => f(a).get(k).map((k, _)) }
+
+      override def map[A, B](fa: SortedMap[K, A])(f: A => B): SortedMap[K, B] =
+        fa.map { case (k, a) => (k, f(a)) }
+
+      override def map2[A, B, Z](fa: SortedMap[K, A], fb: SortedMap[K, B])(f: (A, B) => Z): SortedMap[K, Z] =
+        if (fb.isEmpty) SortedMap.empty(Order[K].toOrdering) // do O(1) work if fb is empty
+        else fa.flatMap { case (k, a) => fb.get(k).map(b => (k, f(a, b))) }
+
+      override def map2Eval[A, B, Z](fa: SortedMap[K, A], fb: Eval[SortedMap[K, B]])(f: (A, B) => Z): Eval[SortedMap[K, Z]] =
+        if (fa.isEmpty) Eval.now(SortedMap.empty(Order[K].toOrdering)) // no need to evaluate fb
+        else fb.map(fb => map2(fa, fb)(f))
+
+      override def ap[A, B](ff: SortedMap[K, A => B])(fa: SortedMap[K, A]): SortedMap[K, B] =
+        fa.flatMap { case (k, a) => ff.get(k).map(f => (k, f(a))) }(scala.collection.breakOut)
+
+      override def ap2[A, B, Z](f: SortedMap[K, (A, B) => Z])(fa: SortedMap[K, A], fb: SortedMap[K, B]): SortedMap[K, Z] =
+        f.flatMap { case (k, f) =>
+          for { a <- fa.get(k); b <- fb.get(k) } yield (k, f(a, b))
+        }
+
+      def foldLeft[A, B](fa: SortedMap[K, A], b: B)(f: (B, A) => B): B =
+        fa.foldLeft(b) { case (x, (k, a)) => f(x, a)}
+
+      def foldRight[A, B](fa: SortedMap[K, A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+        Foldable.iterateRight(fa.values.iterator, lb)(f)
```

Comment:
I thought this was actually unsafe since it captures a mutable iterator? I think to make it safe, it needs to have ` () => Iterator[A]` which it can call to get a new one each time (which is another way of saying the `Iterable[A]` and it calls `.iterator` internally.

We had some bug related to this in the past. Not sure why we didn't fix `Foldable.iterateRight`. Actually, it was related, but not exactly the same. See #1716.